### ISSUE TITLE
Delete <myproject>.zip from /tmp after load to free space (slim_handler=true)

### ISSUE THREAD

**Issue by olirice** [Miserlou/Zappa#1020]
## Context

 

Currently, when using slim_handler=true, the project source zip file at `/tmp/my_project.zip` is not deleted after it is extracted.

Projects that download additional dependencies (tensorflow/sklearn models, binaries, video, etc) at runtime can burn through the available 500 mb pretty quickly and throw OSError resource exhausted errors.

What are your thoughts on deleting the project zip file after it's been decompressed? 

I'll submit a PR if approved.

**Comment by Miserlou** [comment]
Fantastic idea

**Comment by olirice** [comment]
I changed a bit more than originally proposed.

In case it isn't clear why I didn't just `os.remove('/tmp/project.zip')` at the end of `zappa.handler.load_remote_project_zip` please see below

tl;dr
> The punchline is that you can always deploy your resource files with `zappa deploy` rather than downloading them from s3 in application code.

**Scenario - Deployment source is ~60MB and resources (models, etc) are ~300MB** 
Lets say, the resulting zip file with source and resources came to 350MB

Currently, the deployment would fail because slim_handler tries to store project.zip (350MB) and the unzipped folder (360MB) in `/tmp` (500MB) at the same time. Bolting `os.remove('/tmp/project.zip')` onto the end of `load_remote_project_zip` doesn't help because the failure occurs before we get that far.

The most common way around that issue is to copy the large resource files to s3 independently and write a bunch of boilerplate in your application to put them in the right place when you have a cold start.

As written, #1022 tries to store `project.zip` in-memory (failing over onto `/tmp`). In that case, if the lambda instance has enough memory, you can deploy any zip file that decompresses into < 500MB. The punchline is that you can always deploy your resource files with `zappa deploy` rather than downloading them from s3 in application code.

Additionally, you know in advance if your deployment is going to fit in Lambda. If your local project folder is < 500MB and the instance's memory size > ~530MB, its going to fit.
